src/containers/NFT/NFTTabs/NFTTabs.tsx

```diff
@@ -0,0 +1,64 @@
+import React from 'react'
+import { useParams, useRouteMatch } from 'react-router'
+import './styles.scss'
+import Tabs from '../../shared/components/Tabs'
+import { getBuyNFToffers, getSellNFToffers } from '../../../rippled/lib/rippled'
+import { Offers } from './Offers'
+import { Transactions } from './Transactions'
+
+interface Props {
+  tokenId: string
+}
+
+export const NFTTabs = (props: Props) => {
+  const { id = '', tab = 'transactions' } = useParams<{
+    id: string
+    tab: string
+  }>()
+  const match = useRouteMatch()
+  const { tokenId } = props
+
+  function renderTabs() {
+    const { path = '/' } = match
+    const tabs = ['transactions', 'buy-offers', 'sell-offers']
+    // strips :url from the front and the identifier/tab info from the end
+    const mainPath = `${path.split('/:')[0]}/${id}`
+    return <Tabs tabs={tabs} selected={tab} path={mainPath} />
+  }
+
+  function renderTransactions() {
+    let body
+    switch (tab) {
+      case 'sell-offers':
+        body = (
+          <Offers
+            key="SellOffers"
+            tokenId={tokenId}
+            fetchOffers={getSellNFToffers}
+            offerType="SellOffers"
+          />
+        )
+        break
+      case 'buy-offers':
+        body = (
+          <Offers
+            key="BuyOffers"
+            tokenId={tokenId}
+            fetchOffers={getBuyNFToffers}
+            offerType="BuyOffers"
+          />
+        )
+        break
+      default:
+        body = <Transactions tokenId={tokenId} />
+        break
+    }
+    return <div>{body}</div>
```

Comment:
not a big deal, but in this case I think it would be easier to have the function just `return` what you want instead of setting the back of `body`.

Reply:
Yeah and the the wrapping element be where this function is called.